### ISSUE TITLE
Plugin not found error in multi-tenants environment

### ISSUE THREAD

**Issue by shouah** [cloudfoundry/cli#1247]
We rolled the CLI upgrade from version 6.25.0 to 6.30.0 into our enterprise CI pipelines. When our CI framework executes user tasks that includes CLI commands which invoke a plugin that has been installed under $HOME/.cf/plugins:

cf [plugin_command] ...

we get this unexpected error in 6.30.0 which was working before in 6.25.0:

'xxxx' is not a registered command. See 'cf help -a'

We believe this issue is due to a change implemented in CLI version 6.27.0 (#1137 - plugins, help and other plugin commands now first check environment variables $CF_PLUGINS_HOME, then $CF_HOME and then $HOME for the user's custom plugins storage folder).

1. there is a server package that delivers the CLI and a set of global plugins that we want to make available to every automated tasks
2. we do not have CF_PLUGIN_HOME defined
3. all user CLI commands are executed under a common service account in a given server
4. For every set of (login+target+ ... + logoff) CLI commands our solution set a unique and temporary CF_HOME environment variable. That is necessary in order for multiple CLI commands to be executed under the same service account without conflicting with one another -- e.g. CLI commands executed by task #1 target a given CC API URL / Org / Space whereas the CLI commands executed by task #2 target a different CC API URL / Org / Space, without setting a unique CF_HOME we scale up and end up having confused CLI commands using the wrong context (e.g. CLI in task #2 picking up context from task #1).

It looks like the CLI is stopping the plugin search at CF_HOME if it's defined (which we do here). We need the CLI to continue the search to HOME/.cf if the plugin doesn't found in the CF_HOME.

This is because the plugins exist under HOME/.cf/plugins but not under the temporary and unique CF_HOME/.cf/plugins that get created before every task and deleted afterward.

## Platform & Shell Details  

RHEL 6.x using BASH. Our CI solution is IBM UrbanCode Deploy (UCD) using the "CF plugin for UCD".

**Comment by dkoper** [comment]
Hi @shouah 

Why can't you set `CF_PLUGIN_HOME` so the CLI can find the plugin folder?
The change was made to address a bug and simplify a complex set of rules where the CLI would look for the CLI plugins. With the current behaviour, you can either set `$CF_PLUGIN_HOME`  to tell the CLI explicitely where to find them, or let it search in its default location in a folder under where the CLI config file is stored.

Regards,
Dies Koper
CF CLI PM

**Comment by shouah** [comment]
Several reasons why it's not practical:

a) product limitation
1. there is no capability in the IBM plugin for CF CLI to set the CF_PLUGIN_HOME variable on a specific step of a workflow
2. there is no capability in the IBM plugin for CF CLI to set a global environmental variable for a particular execution agent server -- which is needed here. We've submitted a feature request but not sure if it will be accepted or when it will be implemented. For reference: it was extremely difficult to get IBM to implement support for auto-generating a unique CF_HOME for each of the plugin step (basically a product design change) but since this made their product single threaded then that was a good argument however here it isn't as it's not an IBM product impact.

b) user experience
1. we cannot ask our users to set that variable for the simple reason that they are not supposed to nor need to know where the CF_PLUGIN_HOME is located.
2. the IBM plugin setup login, target and logoff: it should "just work" when the user drops an action onto the process design UI.

bottom line:
the users should not need to know how to use the CLI, hence here the need to set CF_PLUGIN_HOME to overwrite the auto-generated CF_HOME variable
